src/nvim/misc1.c

```diff
@@ -2408,10 +2412,24 @@ get_number (
   if (msg_silent != 0)
     return 0;
 
+  cmd_silent = false;  // Want to see the prompt.
+  cmdline_row = msg_row;
+
+  select = getcmdline_prompt('#', prompt, 0, EXPAND_NOTHING, NULL,
+                             input_callback);
+  p = select;
+
+  // Since the user typed this, no need to wait for return.
+  need_wait_return = false;
+  msg_didout = false;
+  cmd_silent = cmd_silent_save;
+  ex_normal_busy = save_ex_normal_busy;
+  callback_free(&input_callback);
+
   no_mapping++;
-  for (;; ) {
-    ui_cursor_goto(msg_row, msg_col);
-    c = safe_vgetc();
+  while (select) {
```

Comment:
Most handling of special chars can be removed from this loop.

We probably still want to support mouse in this mode. Look at `case K_MOUSELEFT` in `command_line_handle_key`, where `#` mode should be special cased. (replacing the text with the number printed as a string is fine)

Reply:
I removed the redundant handling, but how do I make the `#` mode special case in `command_line_handle_key`? Do I add a condition in this:
```c
if (!mouse_has(MOUSE_COMMAND)) {
  return command_line_not_changed(s);
}
```

Reply:
Good question, previously mouse works in `inputlist()` if the mode from which it was called was included in `set mouse`, but now it will be controlled by if `mouse=c` (or `a`, which is the default) is included. We could add an exception so that mouse is always active with `firstc='#'` unless `mouse` is completely empty. (this will also require change to `setmouse()`). For now, you can start with it working with `mouse=c` or `a`, so put the condition _after_ this if block.

if `s->firstc == '#'` you will need to check if the mouse is above the cmdline row, and then `return 0` to abort the cmdline. But you need to somehow signal back to  `get_number` that mouse was used (so that it reads `mouse_row` instead), as `getcmdline_prompt` is not called much, adding a `bool *` out arg would be fine.

Reply:
Hi, sorry for the late reply (acads in college), how do I test if mouse is working? I tried to use mouse to select options in Kitty, Gnome Terminal and Tilda but it did not work in any (even in the master branch)... Which terminal can I use to test mouse?

Reply:
What does `set mouse` give? Try `set mouse=a` . The two first I know should work, if still not, what is `$TERM` ?

Reply:
Ok! So so sorry for this... I had not set mouse. :sweat_smile: